pkg/spanfilter/splitpolicy.go

```diff
@@ -0,0 +1,115 @@
+package spanfilter
+
+import (
+	"fmt"
+
+	"github.com/grafana/tempo/pkg/spanfilter/config"
+	"github.com/grafana/tempo/pkg/spanfilter/policymatch"
+	v1 "github.com/grafana/tempo/pkg/tempopb/resource/v1"
+	tracev1 "github.com/grafana/tempo/pkg/tempopb/trace/v1"
+	"github.com/grafana/tempo/pkg/traceql"
+	"golang.org/x/exp/maps"
+)
+
+// splitPolicy is the result of parsing a policy from the config file to be
+// specific about the area the given policy is applied to.
+type splitPolicy struct {
+	// ResourceMatch is a set of resource attributes that must match a span for the span to match the policy.
+	ResourceMatch *policymatch.AttributePolicyMatch
+	// SpanMatch is a set of span attributes that must match a span for the span to match the policy.
+	SpanMatch *policymatch.AttributePolicyMatch
+	// IntrinsicMatch is a set of intrinsic attributes that must match a span for the span to match the policy.
+	IntrinsicMatch *policymatch.IntrinsicPolicyMatch
+}
+
+func newSplitPolicy(policy *config.PolicyMatch) (*splitPolicy, error) {
```

Comment:
i find this a bit tough to read. some suggestions on cleanup below. some basic tests to confirm that a config becomes the expected split policy would probably be good too.